### PR TITLE
feat:added user-profile page and follow/unfollow functionality

### PR DESCRIPTION
**Pull request by Hardiegogo**
- Added user-profile page
- Added Edit user modal
- Added follow/unfollow feature
- Based on following posts show up in the feed

Screenshots:
image
image
image

### CODE REVIEW

File: src/components/Suggestionbar.jsx [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
+            }else{
+                dispatch(followUser({_id,setIsFollowLoading}))
+            }
+        }
+
+    }
     return <div className='flex justify-between flex-wrap gap-1'>
         <div className='flex items-center font-sans gap-1'>
-            <img className='w-12 rounded-full object-cover h-12 ' src={userProfile} alt="" />
+            <Link to={`/user/${_id}`}><img className='w-12 rounded-full object-cover h-12 ' src={userProfile} alt="" /></Link>
```
Comment (AmishMishra11) [inline src/components/Suggestionbar.jsx:30]: You should add alt here

File: src/utils/user-utils/user-services.js [hunk 2]
```diff
@@ -10,13 +17,75 @@ export const getUsers = createAsyncThunk("users/getUsers", async () => {
...
+export const followUser=createAsyncThunk("users/followUser",async({_id,setIsFollowLoading})=>{
+  const encodedToken=localStorage.getItem('token')
+  try {
+    const res=await axios({
+      method:"POST",
+      url:`/api/users/follow/${_id}`,
+      headers:{authorization:encodedToken}
+    })
+    if(res.status===200){
+      setIsFollowLoading(false)
```
Comment (AmishMishra11) [inline src/utils/user-utils/user-services.js:41]: instead of doing this you should properties of create async thunk and set loading like that

**Review by AmishMishra11 — APPROVED** [review]
Good folder structure
UI looks great

**Review by dev-enforced — APPROVED** [review]
The live preview of Mircle Media is working well with the follow and unfollow functionalities.Code is also good and easy to understand!! Do remove the debuggers consoles that you have added for the thunks and while dispatching them.

**Pull request merged**